_posts/2016-11-14-JS-Glossary.md

```diff
@@ -0,0 +1,554 @@
+
+Yes, the "state" and "state" thing is a little confusing, we'll get to it.
+
+### Props
+
+Props are chunks of app state that are passed into your component from a parent component. In [JSX](#jsx) this is represented as an XML attribute.
+
+Let's check out [an example][jsx-example]:
+
+```js
+export default class Header extends React.Component {
+  [...]
+  render() {
+    return (
+        <View style={styles.followButton}>
+            <InvertedButton text={this.state.following ? 'Following' : 'Follow'}
+                            selected={this.state.following}
+                            onPress={this.handleFollowChange} />
+        </View>
+    )
+  }
+}
+```
+
+See the `InvertedButton` component, it has three `props` being passed in: `text`, `selected` and `onPress`. If any of those props were to change the entire `InvertedButton` component would be re-rendered to the native view hierarchy. These `props` are the key to passing data downwards through your hierarchy. Note: you cannot access the parent component (without passing it in as a prop.)
+
+You should therefore consider `props` as immutable bits of app state relevant to the component it's being passed into. 
+
+### State-again
+
+A component also has a `state` attribute. The key to understanding the difference between `props` and `state` is: `state` is something controlled within that component that can change - `props` do not. 
+
+The above example is a pretty good example of this, when this component is first added to the hierarchy, we send a networking request to get whether you are following something or not. The parent component (`Header`) does not need to update when we know whether you are following or not, but the `InvertedButton` does. So, it is `state` for the parent, but a `prop` for the `InvertedButton`. This means changing the state for `following` will only cause a re-render in the button.
+
+So state is something which changes within a component, which _could_ be used as `props` for it's children. Examples of this are around handling animation progress, whether you're following something, selection indices and any kind of networking which we do outside of [Relay](#relay).
+
+If you'd like to read more, there is a much deeper explanation in [uberVU/react-guide][react-guide]   
+
+### Context
+
+[The docs][context_docs] are pretty specific about context:
+
+> If you aren't an experienced React developer, don't use context. There is usually a better way to implement functionality just using props and state.
+
+Seems to be you should only be using this is really, really specific places. If you need it, you don't need this glossary.
```

Comment:
`Seems to be you should only be using` -> `Seems to be something that you should only be using` ?